openpype/hosts/maya/plugins/publish/extract_camera_mayaScene.py

```diff
@@ -172,18 +172,24 @@ def process(self, instance):
                                                       dag=True,
                                                       shapes=True,
                                                       long=True)
+
+                    attrs = {"backgroundColorR": 0.0,
+                             "backgroundColorG": 0.0,
+                             "backgroundColorB": 0.0,
+                             "overscan": 1.0}
+
                     # Fix PLN-178: Don't allow background color to be non-black
-                    for cam in cmds.ls(
+                    for cam, (attr, value) in itertools.product(cmds.ls(
                             baked_camera_shapes, type="camera", dag=True,
-                            shapes=True, long=True):
-                        attrs = {"backgroundColorR": 0.0,
-                                 "backgroundColorG": 0.0,
-                                 "backgroundColorB": 0.0,
-                                 "overscan": 1.0}
-                        for attr, value in attrs.items():
-                            plug = "{0}.{1}".format(cam, attr)
-                            unlock(plug)
-                            cmds.setAttr(plug, value)
+                            shapes=True, long=True), attrs.items()):
+                        # the above call still pull in shapes that are not
+                        # cameras, so we filter them out here
+                        if cmds.nodeType(cam) != "camera":
+                            continue
+                        plug = "{0}.{1}".format(cam, attr)
+                        unlock(plug)
+                        cmds.setAttr(plug, value)
```

Comment:
that check there was because I've overlooked the `shape` flag and was wondering why the hell I still get other shapes than camera. Apart from that, the code is maybe harder to read but is getting rid of one loop and re-setting of `attrs` that is constant in each iteration.